### PR TITLE
add available languages to cask info command

### PR DESCRIPTION
**Pull request by yzguy**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

-----
This is first ever OSS pull request to a real project, I figured it was a relatively small easy one.

This adds a new method to the DSL parsing to expose the languages specified in a cask, and a new section in the info output for available languages

```
-> brew cask info firefox
firefox: 55.0.3
https://www.mozilla.org/firefox/
Not installed
From: https://github.com/caskroom/homebrew-cask/blob/master/Casks/firefox.rb
==> Name
Mozilla Firefox
==> Languages
cs, de, en-GB, en, es-AR, es-CL, es-ES, fi, fr, gl, in, it, ja, nl, pl, pt, pt-BR, ru, uk, zh-TW, zh
==> Artifacts
Firefox.app (App)
```

The section does not show up if there are no languages specified in the cask

```
-> brew cask info 1password
1password: 6.8.2
https://1password.com/
Not installed
From: https://github.com/caskroom/homebrew-cask/blob/master/Casks/1password.rb
==> Name
1Password
==> Artifacts
1Password 6.app (App)
```

Fixes https://github.com/caskroom/homebrew-cask/issues/31128

I'm absolutely open to any help, suggestions, criticism.

### CODE REVIEW

**Comment by yzguy** [comment]
Absolutely. If you don't mind, I'd like to bounce my thoughts off you before I tackle it. Testing takes a little for me to wrap my head around what is a good test.

By changing `Library/Homebrew/cask/lib/hbc/cli/info.rb`, so we would want a test in `test/cask/cli/info_spec.rb` and by changing `Library/Homebrew/cask/lib/hbc/dsl.rb`, we would want to add a test in `test/cask/dsl.rb`.

For `test/cask/cli/info_spec.rb`, we want a test to check if that a cask specified multiple languages, it would print them, then another checking that nothing is printed if it did not specify any.

For `test/cask/dsl.rb`, we want a test under the `describe "language stanza" do` that checks if any language blocks are specified, we get a flat array of them. If none are specified, we get an empty array.

**Comment by reitermarkus** [comment]
Yes, these sound exactly like the tests we'd need. 👍

**Comment by yzguy** [comment]
Tests added, brew tests on my machine passed, TravisCI tests looked to have passed to. I think that should be everything :)

File: Library/Homebrew/test/cask/dsl_spec.rb [hunk 1]
```diff
@@ -177,6 +177,36 @@
       expect(cask.call.sha256).to eq("xyz789")
       expect(cask.call.url.to_s).to eq("https://example.org/en-US.zip")
     end
+
+    it "returns empty array if no languages specified" do
```
Comment (reitermarkus) [inline Library/Homebrew/test/cask/dsl_spec.rb]: **an** empty array

File: Library/Homebrew/test/cask/cli/info_spec.rb [hunk 2]
```diff
@@ -90,6 +90,38 @@
...
+      ==> Name
+      None
+      ==> Languages
+      zh, en-US
+      ==> Artifacts
+      Caffeine.app (App)
+    EOS
+  end
+
+  it 'should not print "Languages" section divider if the languages block has no output' do
```
Comment (reitermarkus) [inline Library/Homebrew/test/cask/cli/info_spec.rb]: does not print

File: Library/Homebrew/test/cask/cli/info_spec.rb [hunk 3]
```diff
@@ -90,6 +90,38 @@
     EOS
   end
 
+  it "should print languages if the Cask provided any" do
```
Comment (reitermarkus) [inline Library/Homebrew/test/cask/cli/info_spec.rb]: prints

File: Library/Homebrew/test/support/fixtures/cask/Casks/with-conditional-languages.rb [hunk 4]
```diff
@@ -0,0 +1,9 @@
+cask 'with-conditional-languages' do
```
Comment (reitermarkus) [inline Library/Homebrew/test/support/fixtures/cask/Casks/with-conditional-languages.rb]: `without-languages`

**Comment by reitermarkus** [comment]
Great work, @yzguy!

**Comment by yzguy** [comment]
Thank you!

**Pull request merged**